src/app/files.js

```diff
@@ -0,0 +1,104 @@
+'use strict'
+
+var EventManager = require('../lib/eventManager')
+
+function Files (storage) {
+  var event = new EventManager()
+  this.event = event
+  var readonly = {}
+
+  this.exists = function (path) {
+    // NOTE: ignore the config file
+    if (path === '.browser-solidity.json') {
+      return
+    }
+
+    if (this.isReadOnly(path)) {
+      return true
+    } else {
+      return storage.exists(path)
+    }
+  }
+
+  this.get = function (path) {
+    // NOTE: ignore the config file
+    if (path === '.browser-solidity.json') {
+      return
+    }
+
+    if (this.isReadOnly(path)) {
+      return readonly[path]
+    } else {
+      return storage.get(path)
+    }
+  }
+
+  this.set = function (path, content) {
+    // NOTE: ignore the config file
+    if (path === '.browser-solidity.json') {
+      return
```

Comment:
One more thing to consider: `localStorage.setItem` doesn't return a value, but throws an exception if a fault happens. We could either catch that in the Storage API and return true/false or catch it here.